File: chapter_4/3_remove_non_alphabet_characters.py
"""
3. Remove non alphabet characters
Design a Python program that will ask the user to enter a string. The program will remove all non-alphabet characters
(digits and symbols) and display the result.
"""

# Get a string from the user
string = input("Enter a string: ")
# The filtered string will contain a string with alphabets only
filtered_string = ""

# For every character in the string entered by the user
for character in string:
    # Check if the current character is an alphabet
    # If it is an alphabet, add it to the filtered string
    # Otherwise, skip it.
    if (character >= "a" and character <= "z") or (
        character >= "A" and character <= "Z"
    ):  # character.isalpha()
        filtered_string += character
print("Filtered String:", filtered_string)

File: exercises/C/chapter_4 - Strings/3_remove_non_alphabet_characters.c
// 3. Remove non alphabet characters
// Design a Python program that will ask the user to enter a string. The program will remove all non-alphabet characters
// (digits and symbols) and display the result.
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

#define BUF_LEN 50

int main(int argc, char const *argv[])
{
    char string[BUF_LEN], *filtered_string;
    int index = 0, length;

    printf("Enter a string: ");
    fgets(string, BUF_LEN, stdin);
    string[strcspn(string, "\n")] = '\0'; // Remove newline

    length = strlen(string);
    filtered_string = malloc(sizeof(char) * (length + 1)); // Allocate memory
    if (filtered_string == NULL)
    {
        printf("Memory Allocation Failed.\n");
        return 1;
    }

    for (int i = 0; i < length; i++)
    {
        if (isalpha(string[i])) // Keep only alphabetic characters
        {
            filtered_string[index++] = string[i];
        }
    }
    filtered_string[index] = '\0'; // Null-terminate the filtered string

    printf("Filtered String: %s\n", filtered_string);

    free(filtered_string); // Free allocated memory
    return 0;
}
